app/src/main/java/com/github/se/eventradar/viewmodel/ChatViewModel.kt

```diff
@@ -70,6 +70,7 @@ constructor(
         val userId = userRepository.getCurrentUserId()
 
         if (userId is Resource.Success) {
+          observeMessages(userId.data, opponentId)
           it.copy(userId = userId.data)
         } else {
           Log.d(
```

Comment:
It is unconventional to have all of this logic in the `update` block. To separate the act of updating the UI state and getting the user ID, this logic should be outside of the `update` block. If there is a particular reason that you are doing this please let me know (and leave a comment in the code if applicable)

Reply:
Do you mean init block?